pennylane/templates/embeddings/iqp.py

```diff
@@ -0,0 +1,273 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the ``IQPEmbedding`` template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from collections import Sequence
+from itertools import combinations
+
+from pennylane.templates.decorator import template
+from pennylane.ops import RZ, CNOT, Hadamard
+from pennylane.templates import broadcast
+from pennylane.templates.utils import (
+    check_shape,
+    check_wires,
+    check_type,
+    get_shape,
+    check_no_variable,
+)
+
+
+@template
+def zz(parameter, wires):
+    """Template for decomposition of ZZ coupling.
+
+    Args:
+        parameter (float): parameter of z rotation
+        wires (list[int]): qubit indices that the template acts on
+    """
+    CNOT(wires=wires)
+    RZ(2 * parameter, wires=wires[1])
+    CNOT(wires=wires)
```

Comment:
You could use `qml.MultiRZ` here. It will decompose like this template if it's not supported on the device.

Reply:
Nice one, didn't know!

Reply:
We could also update the QAOA embedding!

Reply:
Shall we not add ``MultiRZ`` to the ``Operations`` overview of the docs? This is where I always look for gates...

Reply:
Oops, good catch @mariaschuld. We will need to add the DiagonalQubitUnitary and PauliWord gates as well.

Hopefully once our developer guide is built out, this will help provide a checklist to make sure these aren't missed.

Reply:
I made a separate PR for the QAOAEmbedding :)